### PR TITLE
fix egctl does not support custom controller name.

### LINKED ISSUE #3140: egctl does not support custom controller name
using custom envoy-gateway-config with custom controller name , but the egctl just support default controller name

### PR DESCRIPTION
**Pull request by qicz**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #3140

### CODE REVIEW

File: internal/cmd/egctl/translate.go [hunk 1]
```diff
@@ -276,7 +276,7 @@ func translateGatewayAPIToIR(resources *gatewayapi.Resources) (*gatewayapi.Trans
 	}
 
 	t := &gatewayapi.Translator{
-		GatewayControllerName:   egv1a1.GatewayControllerName,
+		GatewayControllerName:   string(resources.GatewayClass.Spec.ControllerName),
```
Comment (arkodg) [inline internal/cmd/egctl/translate.go:279]: probably good to check if `resources.GatewayClass` is `nil` or not, before dereferncing it

Comment (qicz) [reply internal/cmd/egctl/translate.go:279]: https://github.com/envoyproxy/gateway/blob/67008f83dc7e10df2b5c94c578a9379c3b688be7/internal/cmd/egctl/translate.go#L691-L694

did it

Comment (arkodg) [reply internal/cmd/egctl/translate.go:279]: won't this crash if no `GatewayClass` is provided ?

Comment (qicz) [reply internal/cmd/egctl/translate.go:279]: @arkodg yes, gateway class checked before translate  resouce 
https://github.com/envoyproxy/gateway/blob/67008f83dc7e10df2b5c94c578a9379c3b688be7/internal/cmd/egctl/translate.go#L332-L334

**Review by arkodg — APPROVED** [review]

**Review by shawnh2 — APPROVED** [review]

**Pull request merged**